project/src/main/java/com/google/autograder/servlets/CreateNewAssignmentServlet.java

```diff
@@ -0,0 +1,138 @@
+    private static final String HOST_URL = "https://8080-778d1d95-8447-4f8a-990b-b90da194d107.us-east1.cloudshell.dev";
+
+    @Override
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String requestURL = (String) request.getHeader("Referer");
+        int courseIDIndex = requestURL.indexOf("?courseID=") + 10;
+        String courseID = requestURL.substring(courseIDIndex);
+
+        String authorization = API.getCurrentUserAPIAuthorization();
+
+        if (authorization != null) {
+            try {
+                byte[] postBodyData = buildPostBody(request).getBytes(StandardCharsets.UTF_8);
+                HttpURLConnection connection = buildHttpURLConnection(courseID, authorization, postBodyData.length);
+                connection.getOutputStream().write(postBodyData);
+                connection.connect();
+
+                int responseCode = connection.getResponseCode();
+
+                if (responseCode == 200) {
+                
+                    System.out.println("\n\nBING\tBANG\tBONG\n\n");
+
+                } else {
+
+                    System.out.println("\n\nBAD\tRESPONSE\tCODE:\t" + responseCode + "\n\n");
+
+                }
+
+            } catch(Exception e) {
+                System.out.println(e);
+                e.printStackTrace();
+            }
+        } else {
+            System.out.println("\n" + "INVALID AUTH CODE : " + authorization);
+        }
+
+        response.sendRedirect("/pages/course.html?courseID=" + courseID);
+    }
+
+    private int[] getDayMonthYear(String date) {
+        int firstSeperator = date.indexOf('-');
+        int secondSeperator = date.lastIndexOf('-');
+        int[] values = new int[3];
+
+        values[0] = Integer.parseInt(date.substring(firstSeperator + 1, secondSeperator));
+        values[1] = Integer.parseInt(date.substring(secondSeperator + 1));
+        values[2] = Integer.parseInt(date.substring(0, firstSeperator));
+
+        return values;
+    }
+
+    private int[] getHoursMinutes(String time) {
+        int seperator = time.indexOf(':');
+        int[] values = new int[2];
+
+        values[0] = Integer.parseInt(time.substring(0, seperator));
+        values[1] = Integer.parseInt(time.substring(seperator + 1));
+
+        return values;
+    }
+
+    private String buildPostBody(HttpServletRequest request) throws UnsupportedEncodingException {
+        String description = request.getParameter("description");
+        String maxPoints = request.getParameter("maxPoints");
+        String dueDate = request.getParameter("dueDate");
+        String dueTime = request.getParameter("dueTime");
+        String title = request.getParameter("title");
+
+        int[] dayMonthYear = getDayMonthYear(dueDate);
+        int[] hoursMinutes = getHoursMinutes(dueTime);
+
+        StringBuilder postBody = new StringBuilder();
+
+        postBody.append("{");
```

Comment:
In a separate PR, this should use the Java API (https://developers.google.com/classroom/quickstart/java)

Reply:
OK no problem, working on this currently in another PR.